web/docs/language/features.md

```diff
@@ -1377,20 +1377,69 @@ This function should return the user fields to use when creating a new user upon
 ## Client configuration
 
 You can configure the client using the `client` field inside the `app`
-declaration, 
+declaration,
 
 ```c
 app MyApp {
   title: "My app",
   // ...
   client: {
+    rootComponent: import Root from "@client/Root.jsx",
     setupFn: import mySetupFunction from "@client/myClientSetupCode.js"
   }
 }
 ```
 
 `app.client` is a dictionary with the following fields:
 
+#### `rootComponent: ClientImport` (optional)
+
+`rootComponent` defines the root component of your client application. It is
+expected to be a React component, and Wasp will use it to wrap your entire app.
+It must render its children, which are the actual pages of your application.
+
+You can use it to define a common layout for your application:
+
+```jsx title="src/client/Root.jsx"
+export default async function Root({ children }) {
+  return (
+    <div>
+      <header>
+        <h1>My App</h1>
+      </header>
+      {children}
+      <footer>
+        <p>My App footer</p>
+      </footer>
+    </div>
+  )
+}
+```
+
+You can use it to setup various providers that your application needs:
+
+```jsx title="src/client/Root.jsx"
+import store from './store'
+import { Provider } from 'react-redux'
+
+export default async function Root({ children }) {
+  return (
+    <Provider store={store}>
+      <App>{children}</App>
+    </Provider>
+  )
+}
+
+function App({ children }) {
+  return (
+    <div>
+      <h1>My App</h1>
+      {children}
+    </div>
+  )
+}
```

Comment:
WHy do we need this extra App here? Can't we just have `Root` render children directly in Provider component and that is it?

Reply:
I wanted to offer a more elaborate example where we have the `Provider` wrapping and some `<App />` compnent (which is how you'll see it's done in the Redux docs).

This also works:
```jsx
export default async function Root({ children }) {
  return (
    <Provider store={store}>
      {children}
    </Provider>
  )
}
```

Reply:
Ok! It is good to give exapmles, but I sometimes have the need to reduce them if possible, to make maintainance simpler. But this is probably ok it is not so complex. Unless this might confused them -> do I have to use this App inside this thing? I don't know, your call!

Reply:
I'll provide two use-case examples (Layout + Providers) and a one more elaborate where both are coming together.

Reply:
Updated here: https://github.com/wasp-lang/wasp/pull/1009/commits/207806c7cedd0093bfe9682f4a201d82fff4edd2